src/pages/Transfer/CrossChainTransferForm/index.tsx

```diff
@@ -88,24 +161,36 @@ const CrossChainTransferForm = (): JSX.Element => {
     try {
       setSubmitStatus(STATUSES.PENDING);
 
-      if (!api) return;
-
-      // We can use if else here as we only support two chains
-      if (fromChain === ChainType.RelayChain) {
-        await transferToParachain(
-          api,
-          selectedAccount.address,
-          destination.address,
-          newMonetaryAmount(data[TRANSFER_AMOUNT], RELAY_CHAIN_NATIVE_TOKEN, true)
-        );
-      } else {
-        await transferToRelayChain(
-          window.bridge.api,
-          selectedAccount.address,
-          destination.address,
-          newMonetaryAmount(data[TRANSFER_AMOUNT], RELAY_CHAIN_NATIVE_TOKEN, true)
-        );
-      }
+      if (!XCMBridge || !fromChain || !toChain) return;
+
+      const sendTransaction = async () => {
+        const { signer } = await web3FromAddress(selectedAccount.address.toString());
+
+        const adapter = XCMBridge.findAdapter(fromChain.type);
+
+        const apiPromise = (XCMProvider.getApiPromise(fromChain.type) as unknown) as ApiPromise;
+
+        apiPromise.setSigner(signer);
+
+        // TODO: Version mismatch with ApiPromise type. This should be inferred.
+        adapter.setApi(apiPromise as any);
+
+        const transferAmount = new MonetaryAmount(currency, data[TRANSFER_AMOUNT]);
+        const transferAmountString = transferAmount.toString(true);
+        const transferAmountDecimals = transferAmount.currency.decimals;
+
+        // TODO: Transaction is in promise form
+        const tx: any = adapter.createTx({
+          amount: FixedPointNumber.fromInner(transferAmountString, transferAmountDecimals),
+          to: toChain.type,
+          token: currency.symbol,
+          address: destination.address
+        } as CrossChainTransferParams);
+
+        await DefaultTransactionAPI.sendLogged(apiPromise, selectedAccount.address, tx);
```

Comment:
```suggestion
        await DefaultTransactionAPI.sendLogged(apiPromise, selectedAccount.address, tx, undefined, true);
```

sendLogged checks for an expected event to be emitted and has to option to only wait for the tx to be included in a block (~12 secs wait) vs the tx being finalized (~1 minute wait). Since we block UI, I think we should set:
- `undefined` -> no success event. However, if you have an event that is being emitted on the XCM side, it might be nice to check for that. Future improvement, wouldn't do that now.
- `true` -> only wait for inclusion in the next block. This will give the user a far quicker return with pretty much the same guarantees that the tx is going to be included.